spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/AbstractCassandraJoin.scala

```diff
@@ -0,0 +1,186 @@
+    singleKeyCqlQuery.length
+
+    def checkSingleColumn(column: ColumnRef): Unit = {
+      require(
+        primaryKeyColumnNames.contains(column.columnName),
+        s"Can't pushdown join on column $column because it is not part of the PRIMARY KEY"
+      )
+    }
+
+    // Make sure we have all of the clustering indexes between the 0th position and the max requested
+    // in the join:
+    val chosenClusteringColumns = tableDef.clusteringColumns
+      .filter(cc => colNames.contains(cc.columnName))
+    if (!tableDef.clusteringColumns.startsWith(chosenClusteringColumns)) {
+      val maxCol = chosenClusteringColumns.last
+      val maxIndex = maxCol.componentIndex.get
+      val requiredColumns = tableDef.clusteringColumns.takeWhile(_.componentIndex.get <= maxIndex)
+      val missingColumns = requiredColumns.toSet -- chosenClusteringColumns.toSet
+      throw new IllegalArgumentException(
+        s"Can't pushdown join on column $maxCol without also specifying [ $missingColumns ]"
+      )
+    }
+    val missingPartitionKeys = partitionKeyColumnNames -- colNames
+    require(
+      missingPartitionKeys.isEmpty,
+      s"Can't join without the full partition key. Missing: [ $missingPartitionKeys ]"
+    )
+
+    joinColumnNames.foreach(checkSingleColumn)
+    joinColumnNames
+  }
+
+  //We need to make sure we get selectedColumnRefs before serialization so that our RowReader is
+  //built
+  lazy val singleKeyCqlQuery: (String) = {
+    val whereClauses = where.predicates.flatMap(CqlWhereParser.parse)
+    val joinColumns = joinColumnNames.map(_.columnName)
+    val joinColumnPredicates = whereClauses.collect {
+      case EqPredicate(c, _) if joinColumns.contains(c) => c
+      case InPredicate(c) if joinColumns.contains(c) => c
+      case InListPredicate(c, _) if joinColumns.contains(c) => c
+      case RangePredicate(c, _, _) if joinColumns.contains(c) => c
+    }.toSet
+
+    require(
+      joinColumnPredicates.isEmpty,
+      s"""Columns specified in both the join on clause and the where clause.
+          |Partition key columns are always part of the join clause.
+          |Columns in both: ${joinColumnPredicates.mkString(", ")}""".stripMargin
+    )
+
+    logDebug("Generating Single Key Query Prepared Statement String")
+    logDebug(s"SelectedColumns : $selectedColumnRefs -- JoinColumnNames : $joinColumnNames")
+    val columns = selectedColumnRefs.map(_.cql).mkString(", ")
+    val joinWhere = joinColumnNames.map(_.columnName).map(name => s"${quote(name)} = :$name")
+    val limitClause = limit.map(limit => s"LIMIT $limit").getOrElse("")
+    val orderBy = clusteringOrder.map(_.toCql(tableDef)).getOrElse("")
+    val filter = (where.predicates ++ joinWhere).mkString(" AND ")
+    val quotedKeyspaceName = quote(keyspaceName)
+    val quotedTableName = quote(tableName)
+    val query =
+      s"SELECT $columns " +
+        s"FROM $quotedKeyspaceName.$quotedTableName " +
+        s"WHERE $filter $limitClause $orderBy"
```

Comment:
SPARKC-433 Switches limit and orderby clause order because this ordering is incorrect :(